app/views/transactions/index.html.erb

```diff
@@ -34,44 +34,51 @@
+            <%= form.search_field :category_name_or_account_name_or_name_cont, placeholder: "Search transaction by name, category or amount", class: "placeholder:text-sm placeholder:text-gray-500 relative pl-10 w-full border-none rounded-lg", "data-transactions-search-target": "search", 'data-action': "input->transactions-search#search" %>
             <%= lucide_icon("search", class: "w-5 h-5 text-gray-500 ml-2 absolute inset-0 transform top-1/2 -translate-y-1/2") %>
           </div>
-        <% end %>
+        </div>
+        <div>
+          <button class="cursor-not-allowed border border-gray-200 block h-full rounded-lg flex items-center gap-2 px-4">
+            <%= lucide_icon("list-filter", class: "w-5 h-5 text-gray-500") %>
+            <p class="text-sm font-medium text-gray-900">Filter</p>
+          </button>
+        </div>
+        <div>
+          <%= form.select :date, options_for_select([['All', 'all'], ['7D', 'last_7_days'], ['1M', 'last_30_days'], ["1Y", "last_365_days"]], selected: params[:date]), {}, { class: "block h-full w-full border border-gray-200 rounded-lg text-sm py-2 pr-8 pl-2", "data-transactions-search-target": "date" } %>
+          
+          <!-- Hidden fields for date range -->
+          <%= form.hidden_field :date_gteq, value: '', "data-transactions-search-target": "dateGteq" %>
+          <%= form.hidden_field :date_lteq, value: '', "data-transactions-search-target": "dateLteq" %>
+        </div>
       </div>
-      <div>
-        <button class="cursor-not-allowed border border-gray-200 block h-full rounded-lg flex items-center gap-2 px-4">
-          <%= lucide_icon("list-filter", class: "w-5 h-5 text-gray-500") %>
-          <p class="text-sm font-medium text-gray-900">Filter</p>
-        </button>
+    <% end %>
+    <%= turbo_frame_tag "transactions_list" do %>
+      <div class="bg-gray-25 rounded-xl px-5 py-3 text-xs font-medium text-gray-500 flex items-center gap-6 mb-4">
+        <div class="w-96">
+          <p class="uppercase">transaction</p>
+        </div>
+        <div class="grow uppercase flex justify-between items-center gap-5 text-xs font-medium text-gray-500">
+          <p>account</p>
+          <p>amount</p>
+        </div>
       </div>
-      <%= form_with url: "#", method: :get, class: "flex items-center gap-4", html: { class: "" } do |f| %>
-        <%= f.select :period, options_for_select([['7D', 'last_7_days'], ['1M', 'last_30_days'], ["1Y", "last_365_days"], ['All', 'all']], selected: params[:period]), {}, { class: "block h-full w-full border border-gray-200 rounded-lg text-sm py-2 pr-8 pl-2 cursor-not-allowed", onchange: "this.form.submit();", disabled: true } %>
+      <% if @transactions.empty? %>
+        <p class="text-gray-500 py-4">No transactions for this account yet.</p>
```

Comment:
Can we implement the "empty" state shown here in this frame?

https://www.figma.com/file/lonJmVk3HYkwZoIO7xYP2w/Maybe-App-Design-(Community)?type=design&node-id=2030-1201&mode=design&t=b0N2jucjz2pEQzCX-0